ui/app/js/controllers/permissions_controller.js

```diff
@@ -76,50 +75,55 @@ function($scope, $routeParams, $location, $timeout, Permissions, Search, $modal,
   };
   /* End filtering */
 
-  $scope.openUpdateModal = function(user) {
-
-    var modalInstance = $modal.open({
-      templateUrl: 'permissions_modal.html',
-      controller: 'UserPermissionsCtrl',
-      backdrop: 'static',
-      // size: size,  // can be lg or sm
-      resolve: {
-        user: function () {
-          return user;
-        },
-        users: function () {
-          return $scope.users;
+    $scope.openUpdateModal = function(user) {
+      $scope.is_edit = true;
+      var modalInstance = $modal.open({
+        templateUrl: 'permissions_modal.html',
+        controller: 'UserPermissionsCtrl',
+        backdrop: 'static',
+        size: 'md',  // can be lg or md, sm
+        resolve: {
+          users: function () {
+            return $scope.users;
+          },
+          is_edit: function () {
+            return $scope.is_edit;
+          },
+          user: function () {
+            return user;
+          },
         }
-      }
-    });
-
-
-  };
+      });
+    };
   /* End openUpdateModal */
 
-  $scope.openNewModal = function() {
-
-    var modalInstance = $modal.open({
-      templateUrl: 'user_modal.html',
-      controller: 'NewUserCtrl',
-      backdrop: 'static',
-      // size: 'sm',
-      resolve: {
-        users: function () {
-          return $scope.users;
+    $scope.openNewModal = function() {
+      $scope.is_edit = false;
+      var modalInstance = $modal.open({
+        templateUrl: 'permissions_modal.html',
+        controller: 'UserPermissionsCtrl',
+        backdrop: 'static',
+        size: 'md',
+        resolve: {
+          users: function () {
+            return $scope.users;
+          },
+          is_edit: function () {
+            return $scope.is_edit;
+          },
+          user: function () {
+            return $scope.user;
+          },
         }
-      }
-    });
-
-    modalInstance.result.then(function (user) {
-      $scope.users.push(user);
-      $scope.openUpdateModal(user);
-    }, function () {
-      // modal closed
-    });
+      });
+      // $scope.users.push($scope.user);
+      // modalInstance.result.then(function (user) {
+      // }, function () {
+      //   // modal closed
+      // });
```

Comment:
Please remove this code instead of commenting it out.

Reply:
Okay I will. Thanks for pointing it out